cmd/snap-update-ns/utils_test.go

```diff
@@ -219,6 +220,69 @@ func (s *utilsSuite) TestSecureMkdirAllOpenError(c *C) {
 	})
 }
 
+func (s *utilsSuite) TestPlanWritableMimic(c *C) {
+	restore := update.MockReadDir(func(dir string) ([]os.FileInfo, error) {
+		c.Assert(dir, Equals, "/foo")
+		return []os.FileInfo{
+			update.FakeFileInfo("file", 0),
+			update.FakeFileInfo("dir", os.ModeDir),
+			update.FakeFileInfo("symlink", os.ModeSymlink),
+			update.FakeFileInfo("error-symlink-readlink", os.ModeSymlink),
+			// NOTE: None of the filesystem entries below are supported because
+			// they cannot be placed inside snaps or can only be created at
+			// runtime in areas that are already writable and this would never
+			// have to be handled in a writable mimic.
+			update.FakeFileInfo("block-dev", os.ModeDevice),
+			update.FakeFileInfo("char-dev", os.ModeDevice|os.ModeCharDevice),
+			update.FakeFileInfo("socket", os.ModeSocket),
+			update.FakeFileInfo("pipe", os.ModeNamedPipe),
+		}, nil
+	})
+	defer restore()
+	restore = update.MockReadlink(func(name string) (string, error) {
+		switch name {
+		case "/foo/symlink":
+			return "target", nil
+		case "/foo/error-symlink-readlink":
+			return "", errTesting
+		}
+		panic("unexpected")
+	})
+	defer restore()
+
+	changes, err := update.PlanWritableMimic("/foo")
+	c.Assert(err, IsNil)
+	c.Assert(changes, DeepEquals, []*update.Change{
+		// Store /foo in /tmp.snap/foo while we set things up
```

Comment:
Comment has a typo. Should be '/tmp/.snap/foo'

Reply:
Fixed.